src/sql/azdata.proposed.d.ts

```diff
@@ -1325,6 +1325,10 @@ declare module 'azdata' {
 		 * An event called when the InfoBox is clicked
 		 */
 		onDidClick: vscode.Event<void>;
+		/**
+		 * An event called when the Infobox link is clicked
+		 */
+		onLinkClick: vscode.Event<InfoBoxLinkClickEvent>;
```

Comment:
Hmmm, well the process would still be the same. You'd pass in a localized string with the placeholders and then we replace those with the actual link objects.

The idea being is that we don't have to duplicate any specific link logic here - because right now it's using a completely custom created link and so if we fix an issue with the other links we have to remember to fix this as well.

Plus creating child objects and adding them to components is already a well defined pattern.

I think the biggest issue is going to be nesting them nicely, since the hyperlinks have their own styling that may make it more difficult to inline with the text. But I imagine we should be able to have this component override any CSS that it needs to modify from the static hyperlink component to fix any issues there.